components/DynamicMarkdownContent.js

```diff
@@ -0,0 +1,28 @@
+import { useEffect, useState } from 'react'
+import MarkdownContent from './MarkdownContent'
+
+const ERROR_MESSAGE = 'Error fetching contents'
+
+/**
+ * This component downloads markdown from the specified URL and renders it.
+ * If the URL is null or there was an error fetching the url, render an error message.
+ */
+export default function DynamicMarkdownContent ({ title, url }) {
+  const [markdown, setMarkdown] = useState(url ? null : ERROR_MESSAGE)
+
+  useEffect(() => {
+    if (!markdown) {
+      fetch(url)
+        .then(res => res.text())
+        .then(setMarkdown)
+        .catch(() => setMarkdown(ERROR_MESSAGE))
+    }
+  })
```

Comment:
The more "next.js" approach might actually be to use `getStaticProps` on the page `terms.js` itself: https://nextjs.org/docs/basic-features/data-fetching#getstaticprops-static-generation

E.g.,
```javascript
// terms.js
export default function Terms ({markdown}) {
  return (
    <MarkdownContent
      title='Terms & Conditions'
      markdown={markdown}
    />
  )
}

export async function getStaticProps() {
  // MDX text - can be from a local file, database, anywhere
  const markdown = await fetch(process.env.TERMS_CONDITIONS_URL).then(res => res.text())
  return { props: { markdown } }
}
```

Reply:
Good point, I will try that. We need to keep in mind that by statically embedding markdown content, if it changes, a redeployment becomes necessary.

Reply:
It might also be a good idea to just throw an error in getStaticProps if the file is unfetchable. That might prevent the possibility of the error state you built in and might signal to us when there are issues with fetching the md file.

Reply:
By this, do you mean failing the build?

Reply:
Yea, that's what I was thinking.

Reply:
I implemented `getStaticProps` in https://github.com/ibi-group/otp-admin-ui/pull/36/commits/dcb265f3970842842cee823831423e4641adef87. For now, if contents can't be fetched, a message will indicate that, and if no contents was configured a different message is shown to the user. I need to explore failing the build some more.

Reply:
Okay, failing a build if static embedding of URLs is pretty simple: see this build log https://github.com/ibi-group/otp-admin-ui/pull/36/checks?check_run_id=2306310873#step:8:34 and this commit https://github.com/ibi-group/otp-admin-ui/pull/36/commits/504f599454f910d144be1bf61ddf2f59f186820f

Reply:
Actually, see this build log for a failed deployment.